Network/TCP.hs

```diff
@@ -236,15 +236,35 @@ openTCPConnection_ uri port stashInput = do
     -- like this as it just does a once-only installation of a shutdown handler to run at program exit,
     -- rather than actually shutting down after the action
     addrinfos <- withSocketsDo $ getAddrInfo (Just $ defaultHints { addrFamily = AF_UNSPEC, addrSocketType = Stream }) (Just fixedUri) (Just . show $ port)
+
+    let
+      connectAddrInfo a = do
+        s <- socket (addrFamily a) Stream defaultProtocol
```

Comment:
There's [bracketOnError](http://hackage.haskell.org/package/base-4.14.1.0/docs/Control-Exception-Base.html#v:bracketOnError) for this resource pattern